File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int c = 1, n = nums.size();
        int maj = nums[0], j = 1;

        while(j < n){
            if(nums[j] != maj){
                c--;
            }else{
                c++;
            }

            if(c==0){
                maj = nums[j];
                c = 1;
            }

            j++;
        }
    return maj;
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int i = 1;
        int majorityElement = nums[0];
        int count = 1;
        while (i < nums.length){
            if(nums[i] == majorityElement){
                count++;
            }else{
                count--;
                if(count <= 0){
                    majorityElement = nums[i];
                    count = 1;
                }
            }
            i++;
        }
        return majorityElement;
    }
}